specs/electra/beacon-chain.md

```diff
@@ -779,24 +792,32 @@ def process_registry_updates(state: BeaconState) -> None:
             validator.activation_epoch = activation_epoch
 ```
 
-#### New `process_pending_balance_deposits`
+#### New `process_pending_deposits`
 
 ```python
-def process_pending_balance_deposits(state: BeaconState) -> None:
+def process_pending_deposits(state: BeaconState) -> None:
     available_for_processing = state.deposit_balance_to_consume + get_activation_exit_churn_limit(state)
     processed_amount = 0
     next_deposit_index = 0
 
-    for deposit in state.pending_balance_deposits:
+    for deposit in state.pending_deposits:
         if processed_amount + deposit.amount > available_for_processing:
             break
-        increase_balance(state, deposit.index, deposit.amount)
+        if deposit.epoch >= state.finalized_checkpoint.epoch:
```

Comment:
This should also check for the legacy style deposits import to have caught up with the queue.

Otherwise, deposits from the old style (via eth1_data) and from the new queue can get reordered, possibly messing with decentralized staking pools that rely on a linear deposit order.